tests/native/test_integration_native.py

```diff
@@ -10,31 +10,34 @@
 from manticore.binary import Elf, CGCElf
 from manticore.native.mappings import mmap, munmap
 
-DIRPATH = os.path.dirname(__file__)
+from typing import List, Set
```

Comment:
I don't believe `mypy` checks the type annotations in the `tests` directory. If we're going to use type annotations here, could you update `mypy.ini` also, please?

https://github.com/trailofbits/manticore/blob/934a4f0e332dfdbbe225a3f978a508429d5472a9/mypy.ini#L3

Reply:
You're absolutely right. (My editor was running mypy in the file, regardless of the ini) Updating mypy.ini to include `tests` reveals a handful of issues:
```
tests/other/utils/test_config.py:130: error: Name 'test_parse' already defined on line 113
tests/ethereum/test_detectors.py:92: error: Incompatible types in assignment (expression has type "Type[DetectUnusedRetVal]", base class "EthDetectorTest" defined the type as "None")
tests/ethereum/test_detectors.py:112: error: Incompatible types in assignment (expression has type "Type[DetectSuicidal]", base class "EthDetectorTest" defined the type as "None")
tests/ethereum/test_detectors.py:134: error: Incompatible types in assignment (expression has type "Type[DetectExternalCallAndLeak]", base class "EthDetectorTest" defined the type as "None")
tests/ethereum/test_detectors.py:242: error: Incompatible types in assignment (expression has type "Type[DetectEnvInstruction]", base class "EthDetectorTest" defined the type as "None")
tests/ethereum/test_detectors.py:265: error: Incompatible types in assignment (expression has type "Type[DetectDelegatecall]", base class "EthDetectorTest" defined the type as "None")
tests/ethereum/test_detectors.py:304: error: Incompatible types in assignment (expression has type "Type[DetectRaceCondition]", base class "EthDetectorTest" defined the type as "None")
tests/ethereum/test_detectors.py:403: error: Incompatible types in assignment (expression has type "Type[DetectManipulableBalance]", base class "EthDetectorTest" defined the type as "None")
tests/native/test_state.py:213: error: Name 'testContextSerialization' already defined on line 160
tests/native/test_memory.py:1431: error: Not all arguments converted during string formatting
tests/native/test_cpu_manual.py:11: error: Incompatible import of "Memory" (imported name has type "Type[tests.native.mockmem.Memory]", local name has type "Type[manticore.native.memory.Memory]")
tests/native/test_cpu_automatic.py:6: error: Cannot resolve name "solver" (possible cyclic definition)
Found 12 errors in 6 files (checked 171 source files)
```
Let me take a look at these.

Reply:
I opened #1608 for fixing the existing typing issues in `tests`, and having mypy check that directory by default.